### ISSUE TITLE
Suggestion: have all code in src, not in lib folder. Get rid of output (lib/*.js) files

### ISSUE THREAD

**Issue by timnederhoff** [picuscreative/testcafe-reporter-html#9]
Currently the code in the output file `lib/index.js` is more up to date than the source code `src/index.js`. E.g. sorting, test numbers and url to failed tests (errors) are implemented in te output but nog in the source code.
It might be forgotten to commit the last changes in `src/index.js` (and update tests), and to remove `lib/*` from git?
Now, when I checkout, install and build, I get another version than the one published on npmjs.com

**Comment by timnederhoff** [comment]
ah, I see that last commits were done in the `lib/index.js` as well. So can I assume that this is the source file? If so, the tests doen't make any sense as they are testing the results from the `src/index.js` file.

**Comment by davidcunha** [comment]
Hello @timnederhoff 

I'll have a look into this issue. I saw that you that submitted a PR and then you close it.

**Comment by davidcunha** [comment]
@timnederhoff I spotted the issue you mentioned. The previous PR were done with changes that had impact only on lib/*.js. I'm currently updating the dependencies (which are pretty outdated) and fixing this situation. Thanks!

**Comment by timnederhoff** [comment]
cool, thanks! The PR was for a feature request for including screenshots (if enabled in TestCafe), but the tests failed, so I closed it so I can fix and append tests to have the pipeline working again.

**Comment by davidcunha** [comment]
@timnederhoff Thanks! I submitted a PR #10 with some majors changes. Probably it will be merged by tomorrow. Therefore the best option would be for you to review your feature request with the latest changes.

**Comment by timnederhoff** [comment]
Hi @davidcunha that was quick :) I saw that most of the functionality were 'rebased' to the src folder. However, some things I still miss in the source:
* number:
```javascript
// Test Number
this.tableReports += this.indentString('<td>', 2);
this.tableReports += this.currentTestNumber;
this.tableReports += '</td>\n';
```
* Testcount is present, but wasn't. It it desired to include the total test count in every test row?
* Link to error on status === failed:
```javascript
} else if (result === 'failed') {
        this.tableReports += `<a href="#test-${this.currentTestNumber}">failed</a>`;
```
and 
```javascript
this.report += this.indentString(`<h4 id="test-${this.currentTestNumber}">${heading}</h4>\n`);
```
* I think all in `lib/` folder can now be removed from git right? (and have `lib/` added to `.gitignore`)
I worked it out (including other functionalities because I needed them for some work) [here](https://github.com/timnederhoff/testcafe-reporter-html/blob/master/src/index.js) to have an example

**Comment by davidcunha** [comment]
@timnederhoff Thanks. #10 was approved and merged into master (it's not released yet). Please submit a PR with the aferomentioned changes and screenshots feature that I will review and accept it! You can remove the testcount too.

lib contains the output of the build process with babel. If you want to modify the library the src folder is the place to go, if you want to publish it we'll still need the lib.

**Comment by timnederhoff** [comment]
Hi @davidcunha, thanks you for improving the code so much. I created a PR #11 for the screenshots.

**Comment by timnederhoff** [comment]
> lib contains the output of the build process with babel. If you want to modify the library the src folder is the place to go, if you want to publish it we'll still need the lib.

That's right. As long a you run `gulp build` before publishing (and you always should IMHO), then you have always the latest generated version from code in the lib folder. There is no need to commit this back to your source repository (it's in the published package though). I created PR #12 to remove the lib folder and include it in the publishable package (`.npmignore`)

**Comment by davidcunha** [comment]
@timnederhoff Thanks for the PR! It's released in 1.4.0.
